tests/integration/components/top-nav/component-test.js

```diff
@@ -0,0 +1,26 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | top-nav', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function(assert) {
+    // Set any properties with this.set('myProperty', 'value');
+    // Handle any actions with this.set('myAction', function(val) { ... });
+
+    await render(hbs`<TopNav />`);
+
+    assert.equal(this.element.textContent.trim(), '');
+
+    // Template block usage:
+    await render(hbs`
+      <TopNav>
+        template block text
+      </TopNav>
+    `);
+
+    assert.equal(this.element.textContent.trim(), 'template block text');
+  });
+});
```

Comment:
this is one of the perfects of ember, is that it generate a generic test code file for you that simply tests the components, routes, etc... you can also add to this test to ensure specific UI remains in the code... or create unit tests to ensure functionality doesn't change.

Reply:
Okay cool. Yeah, I need to learn testing techniques. Started a little bit from the docs.